drivers/mipi_dbi/mipi_dbi_rpi_pico_pio.c

```diff
@@ -0,0 +1,535 @@
+	if (ret < 0) {
+		LOG_ERR("Failed to configure PIOs");
+		return ret;
+	}
+
+	if (gpio_is_ready_dt(&config->cmd_data)) {
+		ret = gpio_pin_configure_dt(&config->cmd_data, GPIO_OUTPUT_ACTIVE);
+		if (ret < 0) {
+			failed_pin = "cmd_data";
+			goto fail;
+		}
+		gpio_pin_set_dt(&config->cmd_data, 0);
+	}
+	if (gpio_is_ready_dt(&config->cs)) {
+		ret = gpio_pin_configure_dt(&config->cs, GPIO_OUTPUT_ACTIVE);
+		if (ret < 0) {
+			failed_pin = "cs";
+			goto fail;
+		}
+		gpio_pin_set_dt(&config->cs, 0);
+	}
+	if (gpio_is_ready_dt(&config->reset)) {
+		ret = gpio_pin_configure_dt(&config->reset, GPIO_OUTPUT_ACTIVE);
+		if (ret < 0) {
+			failed_pin = "reset";
+			goto fail;
+		}
+		gpio_pin_set_dt(&config->reset, 0);
+	}
+
+	return ret;
+fail:
+	LOG_ERR("Failed to configure %s GPIO pin.", failed_pin);
+	return ret;
+}
+
+static DEVICE_API(mipi_dbi, mipi_dbi_pico_pio_driver_api) = {
+	.reset = mipi_dbi_pico_pio_reset,
+	.command_write = mipi_dbi_pico_pio_command_write,
+	.write_display = mipi_dbi_pico_pio_write_display};
+
+#define PIO_MIPI_DBI_INIT(n)                                                                       \
+	K_MSGQ_DEFINE(dma_msgq##n, sizeof(int), MIPI_DBI_MAX_SPLITS, 4);                           \
+	static const struct mipi_dbi_pico_pio_config mipi_dbi_pico_pio_config_##n = {              \
+		.dev_dma = DEVICE_DT_GET(DT_NODELABEL(dma)),                                       \
+		.dev_pio = DEVICE_DT_GET(DT_INST_PARENT(n)),                                       \
+		.msq_dma = &dma_msgq##n,                                                           \
+		.data = {GPIO_DT_SPEC_INST_GET_BY_IDX_OR(n, data_gpios, 0, {0}),                   \
+			 GPIO_DT_SPEC_INST_GET_BY_IDX_OR(n, data_gpios, 1, {0}),                   \
+			 GPIO_DT_SPEC_INST_GET_BY_IDX_OR(n, data_gpios, 2, {0}),                   \
+			 GPIO_DT_SPEC_INST_GET_BY_IDX_OR(n, data_gpios, 3, {0}),                   \
+			 GPIO_DT_SPEC_INST_GET_BY_IDX_OR(n, data_gpios, 4, {0}),                   \
+			 GPIO_DT_SPEC_INST_GET_BY_IDX_OR(n, data_gpios, 5, {0}),                   \
+			 GPIO_DT_SPEC_INST_GET_BY_IDX_OR(n, data_gpios, 6, {0}),                   \
+			 GPIO_DT_SPEC_INST_GET_BY_IDX_OR(n, data_gpios, 7, {0}),                   \
+			 GPIO_DT_SPEC_INST_GET_BY_IDX_OR(n, data_gpios, 8, {0}),                   \
+			 GPIO_DT_SPEC_INST_GET_BY_IDX_OR(n, data_gpios, 9, {0}),                   \
+			 GPIO_DT_SPEC_INST_GET_BY_IDX_OR(n, data_gpios, 10, {0}),                  \
+			 GPIO_DT_SPEC_INST_GET_BY_IDX_OR(n, data_gpios, 11, {0}),                  \
+			 GPIO_DT_SPEC_INST_GET_BY_IDX_OR(n, data_gpios, 12, {0}),                  \
+			 GPIO_DT_SPEC_INST_GET_BY_IDX_OR(n, data_gpios, 13, {0}),                  \
+			 GPIO_DT_SPEC_INST_GET_BY_IDX_OR(n, data_gpios, 14, {0}),                  \
+			 GPIO_DT_SPEC_INST_GET_BY_IDX_OR(n, data_gpios, 15, {0})},                 \
+		.data_bus_width = DT_INST_PROP_LEN(n, data_gpios),                                 \
```

Comment:
I've seen quite a great way to make this more compact: https://github.com/zephyrproject-rtos/zephyr/blob/96d062add8c559c02d4d21af9c9a120b067b6f92/drivers/auxdisplay/auxdisplay_gpio_7seg.c#L240-L255